test/test_things.js

```diff
@@ -1,18 +1,200 @@
+      server.post('/Things')
+        .send(defaultThingObject)
+// XXX      .expect('Content-Type', /json/)
+        .expect(400)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(400);
+          done();
+        });
+    });
+
+    it('should create a new Thing', done => {
+      server.post('/Things')
+        .send(defaultThingObject)
+// XXX      .expect('Content-Type', /json/)
+        .expect(201)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(201);
+          res.get('Location').should.match(/Things\/\d*$/)
+          done();
+        });
+    });
+  });
+
+  describe('PATCH Things API', () => {
+    beforeEach(done => {
+      db().then(models => {
+        models[resource].create(defaultThingObject).then((createdThing) => {
+          id = createdThing.id;
+          done();
+        });
       });
+    });
+
+    afterEach((done) => {
+      db().then(models => {
+        models[resource].findById(id).then(thing => {
+          thing.destroy().then(() => {
+            done();
+          })
+        });
+      });
+    });
+
+    it('should return an error if no id is provided', done => {
+      server.patch('/' + resource)
+        .send()
+// XXX      .expect('Content-Type', /json/)
+        .expect(404)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(404);
+          done();
+        });
+    });
+
+    it('should update a Thing with id = 1', done => {
+      defaultThingObject.name = 'changed name';
+      server.patch('/' + resource + '(' + id + ')')
+        .send(defaultThingObject)
+// XXX      .expect('Content-Type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(200);
+          res.body.name.should.equal(defaultThingObject.name);
+          done();
+        });
+    });
+  });
+
+  describe('DELETE Things API', () => {
+    beforeEach(done => {
+      db().then(models => {
+        models[resource].create(defaultThingObject).then((createdThing) => {
+          id = createdThing.id;
+          done();
+        });
+      });
+    });
+
+    it('should return an error if no id is provided', done => {
+      server.delete('/' + resource)
+        .send()
+// XXX      .expect('Content-Type', /json/)
+        .expect(404)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(404);
+          done();
+        });
+    });
+
+    it('should delete a Thing with id', done => {
+      server.delete('/' + resource + '(' + id + ')')
+        .send()
+// XXX      .expect('Content-Type', /json/)
+        .expect(204)
+        .end((err, res) => {
+          should.not.exist(err);
+          res.status.should.be.equal(204);
+          res.body.should.be.empty();
+          done();
+        });
+    });
   });
 });
```

Comment:
This is a good start, but I think we can improve the number of tests here. Check what I have so far for the Sensors endpoint https://github.com/mozilla-sensorweb/sensorthings/pull/7/files#diff-1fc4362bb07bc0f9bc75385b74f7a1ebL3

It's ok to do it in a follow up.